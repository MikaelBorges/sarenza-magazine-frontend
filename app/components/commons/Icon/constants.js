export const CHEVRON_RIGHT = 'CHEVRON_RIGHT';
export const CHEVRON_LEFT = 'CHEVRON_LEFT';
export const CHEVRON_DOWN = 'CHEVRON_DOWN';
export const SPINNER = 'SPINNER';
export const CHECKED = 'CHECKED';
export const CLOSE = 'CLOSE';
