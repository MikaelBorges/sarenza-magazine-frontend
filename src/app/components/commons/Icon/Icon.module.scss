.icon {
  vertical-align: middle;
  line-height: 1;
}
