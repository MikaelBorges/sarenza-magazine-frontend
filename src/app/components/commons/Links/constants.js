export const LINK_TYPE = {
  TEXT: 'test',
  BUTTON: 'button',
  OTHER: 'other'
};

export const ARROW_DIRECTION = {
  RIGHT: 'right',
  DOWN: 'down'
};
