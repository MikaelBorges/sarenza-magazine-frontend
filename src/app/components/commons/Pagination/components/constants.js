export const TEXTS = {
  rowsPerPage: 'Lignes par page :',
  pageLabel: 'Page :',
  pageOn: 'sur',
  viewAll: 'Tout voir',
  next: 'Suiv',
  prev: 'Préc',
  nextButton: 'Suivant',
  prevButton: 'Précédent'
};

export const PAGE_RANGES = ['5', '10', '15', '20', '50', '100'];
