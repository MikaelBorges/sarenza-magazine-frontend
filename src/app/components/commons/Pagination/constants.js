export const TEXTS = {
  rowsPerPage: 'Lignes par page :',
  pageLabel: 'Page :',
  pageOn: 'sur',
  viewAll: 'Tout voir',
  next: 'Suiv',
  prev: 'Pr√©c'
};

export const PAGE_RANGES = ['5', '10', '15', '20', '50', '100'];
