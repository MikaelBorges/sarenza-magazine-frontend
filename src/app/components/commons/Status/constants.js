const STATUS_TYPE = {
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error'
};

export default STATUS_TYPE;
