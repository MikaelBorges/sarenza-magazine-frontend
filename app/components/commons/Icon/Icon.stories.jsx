import React from 'react';

import * as ICONS from './constants';
import Icon from './Icon';

export default {
  title: 'Icon',
  component: Icon
};

export const ChevronDown = () => <Icon name={ICONS.CHEVRON_DOWN} />;
export const ChevronLeft = () => <Icon name={ICONS.CHEVRON_LEFT} />;
export const ChevronRight = () => <Icon name={ICONS.CHEVRON_RIGHT} />;
export const Spinner = () => <Icon name={ICONS.SPINNER} />;
export const Checked = () => <Icon name={ICONS.CHECKED} />;
