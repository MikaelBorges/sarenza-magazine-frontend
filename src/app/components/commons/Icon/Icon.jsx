import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import * as ICONS from './constants';
import styles from './Icon.module.scss';
import Checked from './svg/Checked';
import ChevronDown from './svg/ChevronDown';
import ChevronLeft from './svg/ChevronLeft';
import ChevronRight from './svg/ChevronRight';
import Close from './svg/Close';
import Spinner from './svg/Spinner';

const Icon = ({ name, width, height, title, cssClass }) => {
  let IconComponent;
  switch (name) {
    case ICONS.CHEVRON_RIGHT:
      IconComponent = ChevronRight;
      break;
    case ICONS.CHEVRON_LEFT:
      IconComponent = ChevronLeft;
      break;
    case ICONS.CHEVRON_DOWN:
      IconComponent = ChevronDown;
      break;
    case ICONS.SPINNER:
      IconComponent = Spinner;
      break;
    case ICONS.CHECKED:
      IconComponent = Checked;
      break;
    case ICONS.CLOSE:
      IconComponent = Close;
      break;
    default:
      return '';
  }
  return (
    <div className={classnames(styles.icon, cssClass)} style={{ width, height: height || width }}>
      {IconComponent && (
        <IconComponent
          xmlns="http://www.w3.org/2000/svg"
          version="1.1"
          role="img"
          width="100%"
          height="100%"
          aria-labelledby={title}
          title={title}
        />
      )}
    </div>
  );
};

Icon.propTypes = {
  name: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  title: PropTypes.string,
  cssClass: PropTypes.string
};

Icon.defaultProps = {
  name: null,
  width: '32px',
  height: undefined,
  title: '',
  cssClass: null
};

export default Icon;
