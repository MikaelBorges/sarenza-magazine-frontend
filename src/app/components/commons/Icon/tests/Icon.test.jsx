import { render } from '@testing-library/react';
import React from 'react';

import * as ICONS from '../constants';
import Icon from '../Icon';

describe('<Icon name="xxx" />', () => {
  it('name "CHEVRON_RIGHT" should match the expected snapshot', () => {
    const { container } = render(<Icon name={ICONS.CHEVRON_RIGHT} />);
    expect(container).toMatchSnapshot();
  });
  it('name "CHEVRON_LEFT" should match the expected snapshot', () => {
    const { container } = render(<Icon name={ICONS.CHEVRON_LEFT} />);
    expect(container).toMatchSnapshot();
  });
  it('name "CHEVRON_DOWN" should match the expected snapshot', () => {
    const { container } = render(<Icon name={ICONS.CHEVRON_DOWN} />);
    expect(container).toMatchSnapshot();
  });
});
