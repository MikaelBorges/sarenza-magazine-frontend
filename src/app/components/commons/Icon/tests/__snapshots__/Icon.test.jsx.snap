// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon name="xxx" /> name "CHEVRON_DOWN" should match the expected snapshot 1`] = `
<div>
  <div
    class="icon"
    style="width: 32px; height: 32px;"
  />
</div>
`;

exports[`<Icon name="xxx" /> name "CHEVRON_LEFT" should match the expected snapshot 1`] = `
<div>
  <div
    class="icon"
    style="width: 32px; height: 32px;"
  />
</div>
`;

exports[`<Icon name="xxx" /> name "CHEVRON_RIGHT" should match the expected snapshot 1`] = `
<div>
  <div
    class="icon"
    style="width: 32px; height: 32px;"
  />
</div>
`;
